ToDo-Garden/ToDoGardenUI/Sources/ToDoGardenUIComponent/TextFieldStyle.swift

```diff
@@ -0,0 +1,194 @@
+import UIKit
+
+public enum Styled { }
+
+extension Styled {
+  open class UITextField: UIKit.UITextField {
+    public var configuration: Configuration
+    
+    public init(configuration: Configuration) {
+      self.configuration = configuration
+      super.init(frame: .zero)
+      build()
+    }
+    
+    public required init?(coder: NSCoder) {
+      fatalError("init(coder:) has not been implemented")
+    }
+    
+    open override func textRect(forBounds bounds: CGRect) -> CGRect {
+      let rect = super.textRect(forBounds: bounds)
+      return buildLeftImageMargin(rect)
+    }
+    
+    open override func editingRect(forBounds bounds: CGRect) -> CGRect {
+      let rect = super.textRect(forBounds: bounds)
+      return buildLeftImageMargin(rect)
+    }
+    
+    private func buildLeftImageMargin(_ rect: CGRect) -> CGRect {
+      if let leftImage = configuration.leftImage, let trailing = leftImage.trailing {
+        var insets = UIEdgeInsets.zero
+        insets.left = trailing
+        return rect.inset(by: insets)
+      }
+      
+      return rect
+    }
+    
+    open override func leftViewRect(forBounds bounds: CGRect) -> CGRect {
+      var rect = super.leftViewRect(forBounds: bounds)
+      if let leftImage = configuration.leftImage {
+        rect.origin.x = leftImage.leading
+      }
+      
+      return rect
+    }
+    
+    private func build() {
+      clearButtonMode = .whileEditing
+      // MAKRK: - Border
+      if let border = configuration.border {
+        layer.cornerRadius = border.cornerRadius
+        layer.borderWidth = border.lineWidth
+      }
+      
+      // MAKRK: - Left Image
+      if let leftImage = configuration.leftImage {
+        leftView = UIImageView(image: leftImage.image)
+        leftViewMode = .always
+      }
+      
+      // MARK: - Right Image
+      if let rightImage = configuration.rightImage, let button = value(forKeyPath: "_clearButton") as? UIButton {
+        button.setImage(rightImage.tintedImage, for: .normal)
+      }
+      
+      // MARK: - Bottom Line
+      if let color = configuration.bottomLineColor {
+        buildBottomLine(color: color)
+      }
+    }
+    
+    private func buildBottomLine(color: UIColor) {
+      let line = UIView()
+      line.backgroundColor = color
+      line.translatesAutoresizingMaskIntoConstraints = false
```

Comment:
해당 라인을 조금 더 명시적으로 사용할 수 있게 아래의 [extension](https://github.com/ToDo-Garden/ToDo-Garden/blob/a34d078074f2b0fa096b5520da1b2a87d78dabf4/ToDo-Garden/ToDoGardenUI/Sources/ToDoGardenUIComponent/UIView%2BExtension.swift#L11)을 만들어놓았습니다!

```suggestion
      line.usingAutolayout()
```